processing/src/main/java/org/apache/carbondata/processing/loading/converter/impl/RowConverterImpl.java

```diff
@@ -183,6 +236,35 @@ public CarbonRow convert(CarbonRow row) throws CarbonDataLoadingException {
         }
       }
     }
+
+    /* If non schema fields are present, generate the value for them and convert. */
+    if (bNonSchemaPresent) {
```

Comment:
Please add comment for this, because we can handle during first time passing itself, why skip there and handle here ?

Reply:
Added comment in code. Also have modified converter to run the loop for 2. First iteration converts schema columns and second iteration generates and converts for non schema columns.